apps/emqx_utils/src/emqx_utils_maps.erl

```diff
@@ -175,6 +176,20 @@ binary_key_map(Map) ->
 safe_atom_key_map(Map) ->
     convert_keys_to_atom(Map, fun(K) -> binary_to_existing_atom(K, utf8) end).
 
+-spec to_json(map()) -> emqx_utils_json:json_text().
+to_json(M0) ->
```

Comment:
Micro nit: since this is more HTTP API-oriented than for general map manipulation, I'd suggest that it might fit better in `emqx_utils_api`, but we can move it later of course.

Reply:
Will do in the next PR.